### PR TITLE
Handling case where all molecules are invalid in property optimization

### PR DESCRIPTION
**Pull request by jannisborn**
Currently (with the current code on `master`), if one runs the property optimization and the model produces only one *unique* molecule, we get this error:

```sh
torchdrug/torchdrug/tasks/generation.py", line 147, in reinforce_forward
    raise ValueError("Generation results collapse to singleton molecules")
ValueError: Generation results collapse to singleton molecules
```

Moreover, when we have **only** invalid molecules, we get another error:

```sh
torchdrug/torchdrug/tasks/generation.py", line 809, in reinforce_forward
    if graph.num_nodes.max() == 1:
RuntimeError: max(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.

```

Conceptually, the error is basically the same as reported in #83 (which was fixed in #84). But this time, the affected part is the property optimization, not in the training.

**SOLUTION**:
Basically, what I'm doing in this PR is imitating the two return variables of `reinforce_forward` whenever the number of generated molecules is 0 or 1. 

Please have a look @KiddoZhu

### CODE REVIEW

**Comment by jannisborn** [comment]
Hi @KiddoZhu, it would be great if you could have a look at the PR
